src/annotator/anchoring/pdf.js

```diff
@@ -1,44 +1,70 @@
 /* global PDFViewerApplication */
 
-import seek from 'dom-seek';
-
 import RenderingStates from '../pdfjs-rendering-states';
 
-import { BrowserRange } from './range';
-import { toRange as textPositionToRange } from './text-position';
-import { TextPositionAnchor, TextQuoteAnchor } from './types';
+import { TextPosition, TextRange } from './text-range';
+import { TextQuoteAnchor } from './types';
 
 /**
  * @typedef {import('../../types/api').TextPositionSelector} TextPositionSelector
  * @typedef {import('../../types/api').TextQuoteSelector} TextQuoteSelector
+ * @typedef {import('../../types/api').Selector} Selector
  *
  * @typedef {import('../../types/pdfjs').PDFPageView} PDFPageView
  * @typedef {import('../../types/pdfjs').PDFViewer} PDFViewer
  */
 
+/**
+ * @typedef PdfTextRange
+ * @prop {number} pageIndex
+ * @prop {Object} anchor
+ * @prop {number} anchor.start
+ * @prop {number} anchor.end
+ */
+
 // Caches for performance.
 
 /**
- * Map of page index to page text content as a `Promise<string>`.
+ * Map of page index to page text content.
+ *
+ * @type {Record<number,Promise<string>>}
  */
 let pageTextCache = {};
 
 /**
- * 2D map from `[quote][position]` to `{pageIndex, anchor}` intended to optimize
- * re-anchoring of a pair of quote and position selectors if the position
- * selector fails to anchor on its own.
+ * A cache that associates the quote and position from an annotation's
+ * selectors to the resolved location in the PDF.
+ *
+ * This helps with re-anchoring of a pair of quote and position selectors if
+ * the position selector fails to anchor.
+ *
+ * @type {Record<string, Record<number, PdfTextRange>>}
```

Comment:
I've had a stab at rewriting the comment. Does this make more sense?:

```js
/**
 * A cache that maps a `(quote, text offset in document)` key to a specific
 * location in the document.
 *
 * The components of the key come from an annotation's selectors. This is used
 * to speed up re-anchoring an annotation that was previously anchored in the
 * current session.
 *
 * @type {Record<string, Record<number, PdfTextRange>>}
 */
let quotePositionCache = {};
``` 

Ideally the type would be something like `Map<[string, number], PdfTextRange>` but JavaScript compares non-primitive keys by identity rather than value.

Reply:
My human parsing of `@type {Record<string, Record<number, PdfTextRange>>}` is that it's using a `string` as a key for an object that is in turn keyed by a `number`. I don't see the composed/compound key in that type. Help!